### DISCUSSION TITLE
Snap nodes and show helper lines on resize

### DISCUSSION THREAD

**Discussion by goliney** [xyflow/xyflow#4815] [Ideas]
I'm using NodeResizer. When I resize a node, I want it to "snap" to the size of other nodes, similar to the [Helper Lines](https://reactflow.dev/examples/interaction/helper-lines) pro example.

I managed to modify the `NodeDimensionChange` and `NodePositionChange` events fired in the `onNodesChange` callback before applying these changes to nodes and it worked okay. But as soon as I release the mouse, NodeResizer's `onResizeEnd` event is fired and it doesn't take into account the snapped node size or position. The `ResizeParams` are original and do not represent the snapped node.

What are your recommendations for making nodes snap during the resize?

**Comment by chrtze** [comment]
hey @goliney, we don't have a solution to this at the moment. I did try a similar approach as you and ran into the same issues. We will need to have a closer look, I added this issue to our internal board for the pro examples and will post an update here once we have an update.

**Comment by goliney** [comment]
Thank you for the answer.
